crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -49,8 +50,32 @@ mod unit_tests;
 mod error;
 mod types;
 
+/// Dependencies required for app validating an op.
+pub struct ValidationDependencies {
+    /// Missing hashes that are being fetched.
+    pub missing_hashes: HashSet<AnyDhtHash>,
+    /// Dependencies that are missing to app validate an op.
+    pub hashes_missing_for_op: HashMap<DhtOpHash, HashSet<AnyDhtHash>>,
+}
+
+impl Default for ValidationDependencies {
+    fn default() -> Self {
+        ValidationDependencies::new()
+    }
+}
+
+impl ValidationDependencies {
+    pub fn new() -> Self {
+        Self {
+            missing_hashes: HashSet::new(),
+            hashes_missing_for_op: HashMap::new(),
+        }
+    }
+}
+
```

Comment:
This snuck in from another PR and will be used for skipping op validation when dependencies are missing as well as ceasing to re-trigger the workflow as a whole when no further deps have been fetched in a certain period.